packages/sdk/src/assets/bridge-asset.ts

```diff
@@ -0,0 +1,267 @@
+    })
+  }
+
+  protected async _getMessageFee(
+    chainId: number,
+    holographChainId: number,
+    gasLimit: bigint,
+    gasPrice: bigint,
+    bridgeOutPayload: Hex,
+  ) {
+    return (
+      await this._bridge.simulateContractFunction({
+        chainId,
+        functionName: 'getMessageFee',
+        args: [holographChainId, gasLimit, gasPrice, bridgeOutPayload],
+      })
+    ).result as bigint[] //TODO: fix the type return
+  }
+
+  protected async _getValue(
+    chainId: number,
+    holographToChainId: number,
+    destinationGasLimit: bigint,
+    destinationGasPrice: bigint,
+    bridgeRequestPayload: Hex,
+  ) {
+    const fees = await this._getMessageFee(
+      chainId,
+      holographToChainId,
+      destinationGasLimit,
+      destinationGasPrice,
+      bridgeRequestPayload,
+    )
+
+    // fees consist of two parts: HToken fee and lz fee
+    // fees[0] = Htoken fee is the amount that we charge user for making sure operators can get the job done
+    // fees[1] = lz fee is what LayerZero charge for sending the message cross-chain
+    // we add the two fees together into one number
+    let feeValue: bigint = fees[0] + fees[1]
+
+    // for now, to accommodate us time to properly estimate and calculate fees, we add 25% to give us margin for error
+    return feeValue + feeValue * BigInt(1.25)
+  }
+
+  protected async _estimateBridgeOutDestinationGas(
+    chainId: number,
+    toChainId: number,
+    contractAddress: Address,
+    bridgeOutPayload: Hex,
+  ) {
+    let destinationGasPrice: bigint
+    let destinationGasLimit: bigint // = BigInt(0)
+    const gasController = GAS_CONTROLLER.bridgeNft[chainId]
+
+    if (gasController?.gasLimit) {
+      destinationGasLimit = gasController.gasLimit
+    } else {
+      // we call jobEstimator on destination chain
+      // by supplying 10 million gas, we get back result of how much gas is left from simulation
+      // subtract leftover gas from 10 million to know exactly how much gas is used for tx
+
+      const holographToChainId = getNetworkByChainId(toChainId).holographId
+      const gasEstimationHelperAddress: Address = getGasEstimationAddress(toChainId)
+      const maxGasValue: bigint = hexToBigInt(('0x' + 'ff'.repeat(32)) as Hex)
+      const gasLimitStub: bigint = getTestGasLimit(toChainId) // this is a very big gas value so we can use to estimate with
+
+      const bridgeRequestPayloadStub = this._createBridgeOutPayload(
+        chainId,
+        holographToChainId,
+        contractAddress,
+        maxGasValue,
+        maxGasValue,
+        bridgeOutPayload,
+      )
+
+      const consumedGas = BigInt(
+        (
+          await this._operator.simulateContractFunction({
+            chainId,
+            functionName: 'jobEstimator',
+            args: [bridgeRequestPayloadStub],
+            options: {gas: gasLimitStub}, //{account: gasEstimationHelperAddress , gas: TEST_GAS_LIMIT},
```

Comment:
Is there a reason we are not using the `gasEstimationHelperAddress` variable from above?

Reply:
It's giving me TypeScript problems, even though it's supposed to accept this parameter. I'm going to try a bit more. If I'm not able to pass this value, I'll add a @ts-ignore.